### PR TITLE
Update release workflow documentation

### PR DESCRIPTION
**Pull request by prprabhu-ms**

### CODE REVIEW

File: docs/references/creating-a-release.md [hunk 1]
```diff
@@ -10,41 +10,75 @@ We must do due diligence in testing our packages before any new major/minor or b
...
+    TODO: Add image showing workflow and options, like there used to be here.
 
-When ready to release to NPM, a new package is created off the release branch and the release branch is then merged into `main` and deleted. A git tag is added to the released git-version.
+This workflow will:
 
-## Creating a release through GitHub actions (Preferred)
+1. Create a Pull Request to merge the prerelease branch back into the base branch.
+  1. For stable release only: Before creating the Pull Request, @azure/communiation-react will be version bumped once again to a `-beta.0` version (this ensures that we can continue to add changes of `prerelease`) type.
+1. Create a new release branch off of the prerelease branch. This branch will be used for the eventual release, but *it will never be merged back in the base branch*.
+  1. Commit some release preparation modifications that can not be merged back, like choosing the dependency versions that depend on the type of the release.
```
Comment (prprabhu-ms) [inline docs/references/creating-a-release.md:52]: TODO: List all the steps here.

TODO: Include https://github.com/Azure/communication-ui-library/pull/1642

Comment (prprabhu-ms) [reply docs/references/creating-a-release.md:52]: TODO: sync telemetry version and ensure versions match

File: docs/references/creating-a-release.md [hunk 2]
```diff
@@ -10,41 +10,86 @@ We must do due diligence in testing our packages before any new major/minor or b
...
+### Step 1: Create a Pre-release branch
 
-- First land the change as a PR into `main`.
-- Then, cherry-pick the change as a separate PR onto the release branch.
+Use the [create-prerelease-branch](https://github.com/Azure/communication-ui-library/actions/workflows/create-prerelease-branch.yml) github action to trigger the release preparation workflow.
 
-This process has the following benefits:
+1. Options for this workflow:
+    1. Branch - This is the branch that the release will be created from. Default option is from `main`.
+    2. Bump Type - This is the type of release that will be created, the options for this are:
```
Comment (prprabhu-ms) [inline docs/references/creating-a-release.md:23]: A brief description for each -- when to choose this type -- will be helpful.

File: docs/references/creating-a-release.md [hunk 3]
```diff
@@ -10,41 +10,86 @@ We must do due diligence in testing our packages before any new major/minor or b
...
+1. Create a `prerelease/<release-tag>` branch, bump the package version for `@azure/communication-react` as appropriate and collect all change files into a changelog.
+    1. This step will also create a file called `GROOMME.md` which should be deleted when completing the PR back into the pre-release branch to signify that the changelog has been manually groomed.
+1. Create a Pull Request into the `prerelease/<release-tag>` for manually summarizing the changes collected in the changelog. *Merge this PR before going to step 2*.
+
+For example, when creating a release off of `main` tagged `3.3.0`, the following branches and Pull Requests will be created:
+
+```mermaid
+graph LR
+  main[branch: main]
+  prerelease[branch: prerelease/3.3.0<br/>- bump version to 3.3.0<br/>- collect changelog<br/>-trigger string translations]
```
Comment (prprabhu-ms) [inline docs/references/creating-a-release.md:39]: Update diagram to not reference string translations.

File: docs/references/creating-a-release.md [hunk 4]
```diff
@@ -10,41 +10,86 @@ We must do due diligence in testing our packages before any new major/minor or b
...
+
+Use the [create-release-branch](https://github.com/Azure/communication-ui-library/actions/workflows/create-release-branch.yml) github action to trigger the release branch creation workflow.
+
+1. Options for GH Action:
+    1. Pre-release branch - This is the pre-release branch that was created in the previous workflow. This action should be only done after the changelog is groomed and merged back into the pre-release branch to avoid cherry picking.
+
+This workflow will:
+
+1. Create a Pull Request to merge the prerelease branch back into the base branch.
+  1. For stable release only: Before creating the Pull Request, @azure/communiation-react will be version bumped once again to a `-beta.0` version (this ensures that we can continue to add changes of `prerelease`) type.
```
Comment (prprabhu-ms) [inline docs/references/creating-a-release.md:57]: This isn't rendering properly (shows up as a flat list: 
image

File: docs/references/creating-a-release.md [hunk 5]
```diff
@@ -10,41 +10,86 @@ We must do due diligence in testing our packages before any new major/minor or b
...
+1. Create a new release branch off of the prerelease branch. This branch will be used for the eventual release, but *it will never be merged back in the base branch*.
+  1. On the release branch will check what kind of release this is, beta, stable-minor, or stable-patch.
+  2. Based on the release type it will update the sdk versions to the latest stable, or latest beta for Calling and Chat.
+  3. The action will synchronize the package telemetry verions.
+  4. will commit these changes automatically on the release branch.
+  5. The workflow will then hop back to the pre-release branch and create a PR back into main.
```
Comment (prprabhu-ms) [inline docs/references/creating-a-release.md:63]: This last bit is a repetition from the top.

File: docs/references/creating-a-release.md [hunk 6]
```diff
@@ -10,41 +10,86 @@ We must do due diligence in testing our packages before any new major/minor or b
...
+
+1. Options for GH Action:
+    1. Pre-release branch - This is the pre-release branch that was created in the previous workflow. This action should be only done after the changelog is groomed and merged back into the pre-release branch to avoid cherry picking.
+
+This workflow will:
+
+1. Create a Pull Request to merge the prerelease branch back into the base branch.
+  1. For stable release only: Before creating the Pull Request, @azure/communiation-react will be version bumped once again to a `-beta.0` version (this ensures that we can continue to add changes of `prerelease`) type.
+1. Create a new release branch off of the prerelease branch. This branch will be used for the eventual release, but *it will never be merged back in the base branch*.
+  1. On the release branch will check what kind of release this is, beta, stable-minor, or stable-patch.
```
Comment (prprabhu-ms) [inline docs/references/creating-a-release.md:59]: This list is a bit verbose, and like to bitrot as compared to the workflow file. Can you summarize this a bit?

File: docs/references/creating-a-release.md [hunk 7]
```diff
@@ -10,41 +10,86 @@ We must do due diligence in testing our packages before any new major/minor or b
...
+  prerelease -->|Create Branch| pr
+  pr -.-o|Create Pull Request| prerelease
+```
+
+### Step 2: Create release branch
+
+Use the [create-release-branch](https://github.com/Azure/communication-ui-library/actions/workflows/create-release-branch.yml) github action to trigger the release branch creation workflow.
+
+1. Options for GH Action:
+    1. Pre-release branch - This is the pre-release branch that was created in the previous workflow. This action should be only done after the changelog is groomed and merged back into the pre-release branch to avoid cherry picking.
```
Comment (prprabhu-ms) [inline docs/references/creating-a-release.md:52]: We should add before the workflow description that we should manually sync the string translation on this release branch after 5 days. Link to relevant documentation or the string translation workflow.

File: docs/references/creating-a-release.md [hunk 8]
```diff
@@ -10,41 +10,86 @@ We must do due diligence in testing our packages before any new major/minor or b
...
-1. Craft the [`@azure/communication-react` changelog](https://github.com/Azure/communication-ui-library/blob/main/packages/communication-react/CHANGELOG.md).
-    * This should be done as a Pull Request into the release branch.
-    * To create the changelist grab changes from the packlets' changelogs and prune the changelog lines as necessary; ensure the changelog looks good and changelog lines that equate to small PRs for the same feature are combined. For more information see: [Pruning a Changelog](../references/pruning-a-changelog.md).
-    * Example changelog PR: <https://github.com/Azure/communication-ui-library/pull/701>
-1. Ensure the "Before Release" steps of the [release checklist](../references/release-checklist.md) are completed.
+- The release branch never diverges off of `main`. In theory, it is possible to abandon the release branch at any point and create a new one off of `main` without losing work.
+-  All PR reviews happen on `main`, and the cherry-pick PR simply requires a sign-off. This avoids non-trivial merge conflicts when the release branch is eventually merged back into `main`.
 
-### Publishing the package
+## Publishing the package
```
Comment (prprabhu-ms) [inline docs/references/creating-a-release.md:92]: Please update line 107: Don't merge release branch back into `main`. Delete it once all release steps are over.

File: docs/references/release-checklist.md [hunk 9]
```diff
@@ -30,6 +30,7 @@ Before we release a new version or beta version the following checklist should b
   * Ensure the documentation on the public storybook site matches the latest release.
   * Open the storybook site and ensure the console log that lists the npm package version number matches the released package version number.
 * ✅ Post about release on the [internal releases Teams channel](https://teams.microsoft.com/l/channel/19%3ae12aa149c0b44318b245ae8c30365880%40thread.skype/ACS%2520Deployment%2520Announcements?groupId=3e9c1fc3-39df-4486-a26a-456d80e80f82&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47).
+* ✅ Delete the release branch from github wihtout merging back in so to not leave it hanging.
```
Comment (prprabhu-ms) [inline docs/references/release-checklist.md:33]: Can you also update line 15 above. It is wrong. We should never cherry-pick string translations to/from `main`. Instead we should directly sync translations into the release branch using the dedicated GH action.

**Review by prprabhu-ms — COMMENTED** [review]
LGTM. But looks like I'm not allowed to Approve my own PR :D

File: docs/references/creating-a-release.md [hunk 10]
```diff
@@ -10,41 +10,88 @@ We must do due diligence in testing our packages before any new major/minor or b
...
 See the [release checklist](../references/release-checklist.md) for tasks that must be completed before releasing.
 
-### Release branches
+## Release workflow
 
-A release branch is created for each new beta and stable release (alpha version are excluded from this). The release branch is a short lived branch used to ensure high release quality.
+Both beta and stable release follow a two step workflow, aided by github actions.
 
-While the release branch is active, some changes might be merged into the branch (for bug fixes, features deemed necessary for the release). PRs into the release branch should follow this process when possible:
+### Step 1: Create a Pre-release branch
```
Comment (JamesBurnside) [inline docs/references/creating-a-release.md:17]: nit: can you add just below this the purpose of a prerelease branch. It would be good to let the reader know what is overall happening in this step and why

```
A prerelease branch is for x,y,z. It bumps packages, etcx. etc. and is merged back to main before creating the release branch.
```

Or alternatively above this state the steps and why we do them
```
1. Create Pre-Release Branch
  A prerelease branch is for x,y,z. It bumps packages, etcx. etc. and is merged back to main before creating the release branch.
2. Merge release branch back to main
  This is to ensure blah blah
4. Create release branch off main
  This is the branch the release will be created off... etc.
6. etc.
```

File: docs/references/creating-a-release.md [hunk 11]
```diff
@@ -10,41 +10,88 @@ We must do due diligence in testing our packages before any new major/minor or b
...
+* Following the conclusion of this workflow you will need to kick off [the string localization workflow](https://github.com/Azure/communication-ui-library/actions/workflows/run-td-build.yml) on the `pre-release` branch that results from this.
 
-When ready to release to NPM, a new package is created off the release branch and the release branch is then merged into `main` and deleted. A git tag is added to the released git-version.
+### Step 2: Create release branch
 
-## Creating a release through GitHub actions (Preferred)
+Use the [create-release-branch](https://github.com/Azure/communication-ui-library/actions/workflows/create-release-branch.yml) github action to trigger the release branch creation workflow.
 
-### Getting Ready for release
+We should manually sync the string translations to this branch 5 days before release from the pre-release branch that this originates from. So it is important to make sure the [translation workflow](https://github.com/Azure/communication-ui-library/actions/workflows/run-td-build.yml) is triggered with enough time to be completed.
```
Comment (JamesBurnside) [inline docs/references/creating-a-release.md:52]: nit: imperative tense
```suggestion
String translations should have been updated in this branch 5 days before release. If they have not, run [translation workflow](https://github.com/Azure/communication-ui-library/actions/workflows/run-td-build.yml) and restart at Step x after translations are complete (this typically takes 5 days)
```

File: docs/references/creating-a-release.md (hunk 7 above)
Comment (JamesBurnside) [reply docs/references/creating-a-release.md:55]: +1, would be good to see this at the top or in a checklist before starting making these branches so they don't get halfway through this and realize they are 5 days out from releasing

Comment (JamesBurnside) [reply docs/references/creating-a-release.md:55]: nm, i see it has been added!

**Comment by prprabhu-ms** [comment]
Let's merge this today so that @anjulgarg can follow the new release workflow instructions.

File: docs/references/release-checklist.md [hunk 12]
```diff
@@ -11,8 +11,9 @@ Before we release a new version or beta version the following checklist should b
 * ✅ Manual and automated tests pass
   * See [Test Plan](#test-plan) below.
 * ✅ All translations have been updated
+  * String translations will need to be started manually from pre-release branch using [this](https://github.com/Azure/communication-ui-library/actions/workflows/run-td-build.yml) workflow. 
   * Ensure that strings for both components and composites have complete translations.
-  * If necessary, merge translations into `main` and cherry-pick updated translations to release branch.
+  * Once the string translations are completed they will need to be cherry picked into the release branch before publishing.
```
Comment (anjulgarg) [inline docs/references/release-checklist.md:16]: Can we not run string translations directly in pre-release branch and create the release branch 5 days later so that the release branch has correct changes?

**Review by anjulgarg — APPROVED** [review]

File: docs/references/release-checklist.md (hunk 12 above)
Comment (prprabhu-ms) [reply docs/references/release-checklist.md:16]: Can't.

The prerelease branch is merged back into `main`. We should pretty much never merge string translations from a branch to another.
